scripts/buyAuctionLands.js

```diff
@@ -0,0 +1,198 @@
+import { Bounds } from '../shared/map'
+import { buildCoordinate, splitCoodinatePairs } from '../shared/coordinates'
+import { loadEnv, asSafeAction } from './utils'
+
+const log = new Log('buyAuctionLands')
+const DEFAULT_OPTIONS = {}
+const requiredOptionNames = ['parcels', 'manaTokenAddress']
+
+const buyAuctionLands = {
+  addCommands(program) {
+    program
+      .command('buy')
+      .option('--parcels [parcels]', 'List of parcels to deploy. Required')
+      .option(
+        '--manaTokenAddress [manaTokenAddress]',
+        'MANAToken contract address. Required'
+      )
+      .option('--password [password]', 'Password for the account')
+      .option(
+        '--retryFailedTxs',
+        'If this flag is present, the script will try to retry failed transactions'
+      )
+      .action(
+        asSafeAction(async userOptions => {
+          const options = Object.assign(DEFAULT_OPTIONS, userOptions)
+          checkRequiredOptions(options, requiredOptionNames)
+
+          log.info('Connecting to the blockchain')
+          await connectEth()
+
+          const allParcels = options.parcels
+            ? readJSON(expandPath(options.parcels))
+            : []
+
+          log.info('Checking for invalid parcels')
+          const parcels = await getValidParcels(allParcels)
+
+          if (parcels.length === 0) {
+            log.info('No parcels to bid')
+            return
+          }
+
+          if (options.password) {
+            log.info(`Unlocking account ${eth.getAddress()}`)
+            await eth.wallet.unlockAccount(options.password)
+          }
+
+          await bidOnParcels(
+            parcels,
+            options.manaTokenAddress,
+            options.retryFailedTxs
+          )
+        })
+      )
+  }
+}
+
+async function getValidParcels(allParcels) {
+  const parcels = []
+  const landRegistryContract = eth.getContract('LANDRegistry')
+
+  for (const parcel of allParcels) {
+    const { x, y } = parcel
+    const parcelId = buildCoordinate(x, y)
+
+    if (!Bounds.inBounds(x, y)) {
+      log.info(`Parcel ${parcelId} is out of bounds`)
+      continue
+    }
+
+    const owner = await landRegistryContract.ownerOfLand(x, y)
+    if (!Contract.isEmptyAddress(owner)) {
+      log.info(
+        `Parcel ${parcelId} already has an owner on the blockchain: ${owner}`
+      )
+      continue
+    }
+    parcels.push(parcel)
+  }
+
+  return parcels
+}
+
+async function bidOnParcels(parcels, tokenAddress, shouldRetry = false) {
+  const landAuctionContract = eth.getContract('LANDAuction')
+  const account = eth.getAddress()
+
+  const gasPrice = (await landAuctionContract.gasPriceLimit()).toNumber()
+  const landLimit = (await landAuctionContract.landsLimitPerBid()).toNumber()
+  log.info(
+    `Using ${gasPrice} as gasPrice and ${landLimit} as land limit per bid`
+  )
+
+  const txs = []
+  const parcelsToRetry = []
+
+  log.info(`Buying ${parcels.length} parcels with token ${tokenAddress}`)
+  await asyncBatch({
```

Comment:
Can we wait some time for N txs to be confirmed/rejected before sending new ones? This would work as other scripts.

Reply:
Yes! I actually forgot. The script is waiting for each transaction to finish but after sending all transactions